tests/framework/project/test_pipeline_discovery.py

```diff
@@ -0,0 +1,170 @@
+import sys
+import textwrap
+import warnings
+from pathlib import Path
+
+import pytest
+
+from kedro.framework.project import configure_project, find_pipelines
+
+
+@pytest.fixture
+def mock_package_name_with_pipelines(tmp_path, request):
+    package_name = "test_package"
+    pipelines_dir = tmp_path / package_name / "pipelines"
+    pipelines_dir.mkdir(parents=True)
+    (pipelines_dir / "__init__.py").touch()
+    for pipeline_name in request.param:
+        pipeline_dir = pipelines_dir / pipeline_name
+        pipeline_dir.mkdir()
+        (pipeline_dir / "__init__.py").write_text(
+            textwrap.dedent(
+                f"""
+                from kedro.pipeline import Pipeline, node, pipeline
+
+
+                def create_pipeline(**kwargs) -> Pipeline:
+                    return pipeline([node(lambda: 1, None, "{pipeline_name}")])
+                """
+            )
+        )
+    sys.path.insert(0, str(tmp_path))
+    yield package_name
+    sys.path.pop(0)
+
+    # Make sure that the `importlib_resources.files` in `find_pipelines`
+    # will point to the correct `test_package.pipelines` not from cache.
+    del sys.modules[f"{package_name}.pipelines"]
+
+
+@pytest.fixture
+def pipeline_names(request):
+    return request.param
+
+
+@pytest.mark.parametrize(
+    "mock_package_name_with_pipelines,pipeline_names",
+    [(x, x) for x in [set(), {"my_pipeline"}]],
+    indirect=True,
+)
+def test_find_pipelines(
+    mock_package_name_with_pipelines,
+    pipeline_names,
+):
+    configure_project(mock_package_name_with_pipelines)
+    pipelines = find_pipelines()
+    assert set(pipelines) == pipeline_names | {"__default__"}
+    assert sum(pipelines.values()).outputs() == pipeline_names
+
+
+@pytest.mark.parametrize(
+    "mock_package_name_with_pipelines,pipeline_names",
+    [(x, x) for x in [set(), {"good_pipeline"}]],
```

Comment:
Stupid question: why do we need `(x, x)` here? Why can't we just use one argument for both `mock_package_name_with_pipelines` and `pipeline_names`?